src/YoutubeAtom.tsx

```diff
@@ -164,9 +165,50 @@ export const YoutubeAtom = ({
     const [hasUserLaunchedPlay, setHasUserLaunchedPlay] = useState<boolean>(
         false,
     );
+    const [hasUserHovered, setHasUserHovered] = useState<boolean>(false);
     const player = useRef<YoutubePlayerType>();
 
+    const hasOverlay = overrideImage || posterImage;
+
+    /**
+     * Show the overlay if:
+     * - It exists
+     *
+     * and
+     *
+     * - It hasn't been clicked upon
+     */
+    const showOverlay = hasOverlay && !hasUserLaunchedPlay;
+    /**
+     * Show a placeholder if:
+     *
+     * - We don't have an iframe source yet (probably because we don't have consent)
+     *
+     * and
+     *
+     * - There's no overlay to replace it with or the reader clicked to play but we're
+     * still waiting on consent
+     *
+     */
+    const showPlaceholder = !iframeSrc && (!hasOverlay || hasUserLaunchedPlay);
+    /**
+     * Load the you tube iframe if:
+     *
+     * - We have a source string defined (i.e. We have consent)
+     *
+     * and
+     *
+     * - One of these 3 things are true
+     *      - We don't have an overlay - so we have to load the video straight away
+     *      - The user has clicked on the overlay, so load the video iframe!
+     *      - The user has moved their mouse over the overlay so lets pre load
+     *        the content
+     */
+    const loadIframe =
+        iframeSrc && (!hasOverlay || hasUserHovered || hasUserLaunchedPlay);
+
     useEffect(() => {
+        if (!consentState) return;
```

Comment:
Should this also early return, if the iframe src has been set?

Just wondering about where `consentState` is truthy and `hasUserLaunched` toggles from false to true it would set the iframeSrc and reinitialise the player again?

Reply:
This is a really great spot! You're completely right, it's semantically wrong to set the source again if it's already been set.

I _think_ this won't cause double loads because we gate the download using `loadIframe` but the fact I have to say 'think' is a strong indication the code is too open. Fixed 👍

Reply:
Hi @arelra I ended up reverting this change because it blocked the `autoplay` value being set in an  edge case.

If the reader has their mouse cursor over the video whilst it is being hydrated then the `onMouseOver` event isn't fired which means the overlay is removed but the video doesn't start playing 😐 .

Resetting the `iframeSrc` string shouldn't impact anything, it's still only loaded once

Reply:
I've tested this code in isolation (via Storybook) and when loaded into DCR and I'm seeing the expected behaviour in each case.

Next steps:

1) This PR is an improvement to the page load experience in all cases but could be seen to add a delay to when videos are played on mobile. I want to add a check for device and, when mobile, use `onIdle` and/or `onVisible` to trigger the download.

2) Once point 1 is complete we're in a position to create a new build and bring into DCR for more testing on `code`